syntax = "proto3";

message Request{
    oneof req {
        RequestMatricula rm = 1;
        RequestLista rl = 2;
    }
}

message RequestMatricula{
    // 1 -> Inserir Matricula
    // 2 -> Alterar nota
    // 3 -> Alterar falta
    int32 tipoRequest = 1;
    Matricula matricula = 2;
}

message RequestLista{
    // 1 -> Lista Alunos
    // 2 -> Lista disciplina, notas e faltas de um aluno
    int32 tipoRequest = 1;
    int32 ano = 2;
    int32 semestre = 3;
    int32 ra = 4;
    int32 cod_disciplina = 5;
}

message Curso{
    int32 codigo = 1;
    string nome = 2;
}

message Aluno{
    int32 RA = 1;
    string nome = 2;
    int32 periodo = 3;
    int32 cod_curso = 4;    
}

message Matricula{
    int32 RA = 1;
    string cod_disciplina = 2;
    int32 ano = 3;
    int32 semestre = 4;
    float nota = 5;
    int32 faltas = 6;
}

message Disciplina{
    string codigo = 1;
    string nome = 2;
    string professor = 3;
    int32 cod_curso = 4;    
}

message ListaAlunos{
    repeated Aluno alunos = 1;
}

message ReturnListaMatriculas{
    repeated RetornoMatricula matriculas = 1;
    bool sucesso = 2;
    string mensagem = 3;
}

message RetornoMatricula {
    int32 ra = 1;
    string nome_disciplina = 2;
    float nota = 3;
    int32 faltas = 4;
}

message RetornoDefault {
    bool sucesso = 1;
    string mensagem = 2;
}


message ListaCursos{
    repeated Curso cursos = 1;
}

message ReturnListaAlunos{
    bool sucesso = 1;
    string mensagem = 2;
    ListaAlunos listaAlunos = 3;
}